spring-web/src/main/java/org/springframework/http/codec/json/AbstractJackson2Decoder.java

```diff
@@ -174,22 +177,38 @@ protected Flux<DataBuffer> processInput(Publisher<DataBuffer> input, ResolvableT
 	@Override
 	public Mono<Object> decodeToMono(Publisher<DataBuffer> input, ResolvableType elementType,
 			@Nullable MimeType mimeType, @Nullable Map<String, Object> hints) {
-
 		return DataBufferUtils.join(input, this.maxInMemorySize)
-				.flatMap(dataBuffer -> Mono.justOrEmpty(decode(dataBuffer, elementType, mimeType, hints)));
+				.flatMap(dataBuffer -> {
+							try {
+								ObjectReader objectReader = getObjectReader(elementType, mimeType, hints);
+								return customizeReaderFromStream(objectReader, mimeType, elementType, hints)
+										.flatMap(reader -> {
+											try {
```

Comment:
There's some code re-use here - unsure but it may be worth putting this all in it's own private helper function?

Reply:
They do slightly different things and I didn't want to break anything so left it like this.